src/GovUk.Education.ExploreEducationStatistics.Content.Services/Cache/ThemeCacheService.cs

```diff
@@ -0,0 +1,114 @@
+﻿#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Common.Cache;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.Interfaces.Cache;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.Requests;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.ViewModels;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace GovUk.Education.ExploreEducationStatistics.Content.Services.Cache;
+
+public class ThemeCacheService : IThemeCacheService
+{
+    private readonly IThemeService _themeService;
+    private readonly ILogger<ThemeCacheService> _logger;
+
+    public ThemeCacheService(
+        IThemeService themeService, 
+        ILogger<ThemeCacheService> logger)
+    {
+        _themeService = themeService;
+        _logger = logger;
+    }
+
+    [BlobCache(typeof(PublicationTreeCacheKey))]
+    private Task<IList<ThemeTree<PublicationTreeNode>>> GetFullPublicationTree()
+    {
+        return _themeService.GetPublicationTree();
+    }
+    
+    [BlobCache(typeof(PublicationTreeCacheKey), forceUpdate: true)]
+    public Task<IList<ThemeTree<PublicationTreeNode>>> UpdatePublicationTree()
+    {
+        _logger.LogInformation("Updating cached Publication Tree");
+        return _themeService.GetPublicationTree();
+    }
```

Comment:
I think these `BlobCache` attributes are missing a `ServiceName = "target"` attribute to signal that during the get or get and update that it needs to find the appropriate `BlobCacheService` by target service name that's wired up with the a `BlobStorageService` with the **public** storage account.

I think this is only working locally during testing because the storage emulator has a single account that we use for both the private and public account connection strings.